src/main/java/baritone/utils/player/AbstractPlayerContext.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * This file is part of Baritone.
+ *
+ * Baritone is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Baritone is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Baritone.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+package baritone.utils.player;
+
+import baritone.api.utils.BetterBlockPos;
+import baritone.api.utils.IPlayerContext;
+import net.minecraft.block.BlockSlab;
+
+/**
+ * @author Brady
+ * @since 11/12/2018
+ */
+public abstract class AbstractPlayerContext implements IPlayerContext {
```

Comment:
dont get why this intermediary class needs to exist tbh

Reply:
It's to implement ``playerFeet()`` just once, so it'll apply to ``LocalPlayerContext`` and whatever the ``BotPlayerContext`` will be.

Reply:
yeah i'm saying define it as a default in `IPlayerContext`

Reply:
Oh I'm dumb lol!